src/ArrayUtils.f90

```diff
@@ -1372,4 +1393,89 @@ SUBROUTINE replaceEntry_1DString(oldlist,sublist,entry)
 
 ENDSUBROUTINE replaceEntry_1DString
 !
+!-------------------------------------------------------------------------------
+!> @brief Pops an entry from a string list
+!> @param list the original list
+!> @param entry the index of the entry to pop
+!> @returns popped the popped entry of the list
+!>
+FUNCTION popEntry_1DString(list,entry) RESULT(popped)
+  TYPE(StringType),ALLOCATABLE,INTENT(INOUT) :: list(:)
+  INTEGER(SIK),INTENT(IN) :: entry
+  TYPE(StringType) :: popped
+  !
+  TYPE(StringType),ALLOCATABLE :: tmplist(:)
+
+  REQUIRE(ALLOCATED(list))
+  REQUIRE(entry > 0)
+  REQUIRE(entry <= SIZE(list))
+
+  CALL MOVE_ALLOC(list,tmplist)
+  ALLOCATE(list(SIZE(tmplist)-1))
+  list(1:entry-1)=tmplist(1:entry-1)
+  list(entry:SIZE(tmplist)-1)=tmplist(entry+1:)
+  popped=tmplist(entry)
+
+ENDFUNCTION popEntry_1DString
+!
+!-------------------------------------------------------------------------------
+!> @brief Inserts a string into a list of strings
+!> @param list the original list
+!> @param insert the new string to insert
+!> @param entry the index to insert the new string at
+!>
+!> After insertion, the @c entry-th index of the list will be @c insert, with
+!> the original values at @c entry and later being moved back by one.  If
+!> @c entry is given a value of @c SIZE(list)+1, then @c insert is added to the
+!> back of the list
+!>
+SUBROUTINE insertEntry_1DString_0Dentry(list,insert,entry)
+  TYPE(StringType),ALLOCATABLE,INTENT(INOUT) :: list(:)
+  TYPE(StringType),INTENT(IN) :: insert
+  INTEGER(SIK),INTENT(IN) :: entry
+  !
+  TYPE(StringType),ALLOCATABLE :: oldlist(:)
+
+  REQUIRE(ALLOCATED(list))
+  REQUIRE(entry > 0)
+  REQUIRE(entry <= SIZE(list)+1)
+
+  CALL MOVE_ALLOC(list,oldlist)
+  ALLOCATE(list(SIZE(oldlist)+1))
+  list(1:entry-1)=oldlist(1:entry-1)
+  list(entry)=insert
+  list(entry+1:SIZE(list))=oldlist(entry:SIZE(oldlist))
+
+ENDSUBROUTINE insertEntry_1DString_0Dentry
+!
+!-------------------------------------------------------------------------------
+!> @brief Inserts a list of string into a list of strings
+!> @param list the original list
+!> @param insert the new list to insert
+!> @param entry the index to insert the new list at
+!>
+!> After insertion, the @c entry-th index of the list will be the first element
+!> of @c insert, with the original values at @c entry and later being moved back
+!> by one.  If @c entry is given a value of @c SIZE(list)+1, then @c insert is
+!> added to the back of the list
+!>
+SUBROUTINE insertEntry_1DString_1Dentry(list,insert,entry)
+  TYPE(StringType),ALLOCATABLE,INTENT(INOUT) :: list(:)
+  TYPE(StringType),INTENT(IN) :: insert(:)
+  INTEGER(SIK),INTENT(IN) :: entry
```

Comment:
Just so we can be clear on the expected behavior, what happens if `insert` is size 0? I personally believe it should go through this interface and return the original list, but there are a handful of assumptions in that.

Reply:
Talked offline, but for the sake of completeness: that's exactly what happens.  Updated unit tests show that.